packages/studio-be/src/core/bots/bot-service.ts

```diff
@@ -197,6 +232,95 @@ export class BotService {
     await this.mountBot(destBotId)
   }
 
+  async getBotTemplates(): Promise<any[]> {
+    const builtinPath = getBuiltinPath('bot-templates')
+    const templates = await fse.readdir(builtinPath)
+
+    const detailed = await templates.map(id => {
+      try {
+        const details = require(path.join(builtinPath, id, 'bot.config.json'))
+        return { id, name: details.name, desc: details.desc, moduleId: 'builtin', moduleName: 'Botpress Builtin' }
+      } catch (err) {}
+    })
+
+    return detailed.filter(x => x !== undefined)
+  }
+
+  async makeBotId(botId: string, workspaceId: string) {
+    const workspace = await this.workspaceService.findWorkspace(workspaceId)
+    return workspace?.botPrefix ? `${workspace.botPrefix}__${botId}` : botId
+  }
+
+  async addBot(bot: BotConfig, botTemplate: BotTemplate): Promise<void> {
+    const { error } = Joi.validate(bot, BotCreationSchema)
+    if (error) {
+      throw new InvalidOperationError(`An error occurred while creating the bot: ${error.message}`)
+    }
+
+    const mergedConfigs = await this._createBotFromTemplate(bot, botTemplate)
+    if (mergedConfigs) {
+      if (!mergedConfigs.disabled) {
+        await coreActions.onBotCreation(bot.id)
+      }
+
+      this._invalidateBotIds()
+    }
+  }
+
+  private async _createBotFromTemplate(botConfig: BotConfig, template: BotTemplate): Promise<BotConfig | undefined> {
+    const templatePath = path.resolve(getBuiltinPath('bot-templates'), template.id)
+    const templateConfigPath = path.resolve(templatePath, BOT_CONFIG_FILENAME)
+
+    try {
+      const scopedGhost = this.ghostService.forBot(botConfig.id)
+      const files = await this._loadBotTemplateFiles(templatePath)
+
+      if (!(await fse.pathExists(templateConfigPath))) {
+        throw new Error("Bot template doesn't exist")
+      }
+
+      const templateConfig = JSON.parse(await fse.readFile(templateConfigPath, 'utf-8'))
+      const mergedConfigs = {
+        ...DEFAULT_BOT_CONFIGS,
+        ...templateConfig,
+        ...botConfig,
+        version: process.BOTPRESS_VERSION
+      }
+
+      if (!mergedConfigs.imports.contentTypes) {
+        const allContentTypes = await this.cms.getAllContentTypes()
+        mergedConfigs.imports.contentTypes = allContentTypes.map(x => x.id)
+      }
+
+      if (!mergedConfigs.defaultLanguage) {
+        mergedConfigs.disabled = true
+      }
+
+      await scopedGhost.upsertFile('/', BOT_CONFIG_FILENAME, stringify(mergedConfigs))
```

Comment:
On Botpress, we used to also create the folder structure like so:

```js
await scopedGhost.ensureDirs('/', BOT_DIRECTORIES)
```

Shouldn't we do the same here?

Reply:
Not necessary, I tested it, removed that part